### ISSUE TITLE
apt module won't work with build-dep

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29710]
From @antonioantunes on 2016-08-13T16:03:59Z
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

apt
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /Users/ikkizenho/Documents/Ansible/playbooks/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

[defaults]
hostfile = hosts
remote_user = root
remote_port = 60022
private_key_file = ~/.ssh/id_rsa
host_key_checking = False
pipelining = True
gathering = smart
fact_caching_timeout = 86400
fact_caching = jsonfile
fact_caching_connection = /tmp/ansible_fact_cache
forks = 20
##### OS / ENVIRONMENT

Running from: OS X El Capitan 10.11.6
Managing: Debian 8 Jessie
##### SUMMARY

when using apt module with "state=build-dep", it can't find packages that could be installed through "apt-get build-dep package"
##### STEPS TO REPRODUCE

```
- name: install libtorrent dependencies
  apt: pkg=libtorrent-rasterbar state=build-dep update_cache=yes cache_valid_time=3600
```
##### EXPECTED RESULTS

expect to install all these packages:

```
root@galileo:/etc# apt-get build-dep libtorrent-rasterbar
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  debhelper dh-python docutils-common geoip-bin intltool-debian libboost-python-dev libboost-python1.55-dev libboost-python1.55.0 libboost-system-dev libboost-system1.55-dev libboost-system1.55.0
  libboost1.55-dev libexpat1-dev libgeoip-dev libmpdec2 libpython-all-dbg libpython-all-dev libpython-dbg libpython-dev libpython2.7 libpython2.7-dbg libpython2.7-dev libpython3-all-dbg
  libpython3-all-dev libpython3-dbg libpython3-dev libpython3-stdlib libpython3.4 libpython3.4-dbg libpython3.4-dev libpython3.4-minimal libpython3.4-stdlib libssl-dev pkg-config po-debconf
  python-all python-all-dbg python-all-dev python-dbg python-dev python-docutils python-roman python2.7-dbg python2.7-dev python3 python3-all python3-all-dbg python3-all-dev python3-dbg
  python3-dev python3-docutils python3-minimal python3-roman python3.4 python3.4-dbg python3.4-dev python3.4-minimal sgml-base xml-core zlib1g-dev
0 upgraded, 60 newly installed, 0 to remove and 0 not upgraded.
Need to get 97.4 MB/99.7 MB of archives.
After this operation, 340 MB of additional disk space will be used.
```
##### ACTUAL RESULTS

```
TASK [install libtorrent dependencies] *****************************************
fatal: [galileo]: FAILED! => {"changed": false, "failed": true, "msg": "No package matching 'libtorrent-rasterbar' is available"}
```

Copied from original issue: ansible/ansible-modules-core#4420

**Comment by ansibot** [comment]
From @ansibot on 2016-08-13T16:03:59Z



@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @TheSeanBrady on 2016-08-13T16:03:59Z

I have experienced a similar issue, and removing the "build-dep" resolved it. Ansible version 2.1.1.0.

**Comment by ansibot** [comment]
From @abadger on 2016-08-13T16:03:59Z

A fix to the build-dep functionality went into 2.2.0 that may fix this.
The fix was for packages that have a source package but no binary package
of that name.  Could you please give 2.2.0 a try and see if this problem
has been resolved?

needs_info

**Comment by ansibot** [comment]
From @ansibot on 2016-08-13T16:03:59Z

@antonioantunes ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @mgwilliams
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by abadger** [comment]
Since there was a fix in 2.2.0 that is thought to address this and there has been no reply, I'm closing this issue.  If you continue to encounter this problem, please feel free to open a new ticket (referencing this one if you think it is relevant.)
